### ISSUE TITLE
Astroid raises exception parsing multi-line string with f-formatting

### ISSUE THREAD

**Issue by BSchilperoort** [pylint-dev/astroid#2644]
I am not sure what exactly is going wrong, but I encountered a confusing error with the Sphinx AutoAPI which uses astroid.

The code is in the following file (slightly modified to make it more condensed):

```py
from importlib.metadata import entry_points, packages_distributions

_model_entrypoints = entry_points(group="ewatercycle.models")

def get_package_name(module_name: str) -> str:
    if module_name in packages_distributions():
        return packages_distributions()[module_name][0]
    return "[unknown]"

try:
    for _model in _model_entrypoints:
        globals()[_model.name] = _model.load()
except Exception as e:
    pkg_name = get_package_name(_model.value.split(".")[0])
    msg = (
        "An error was raised when trying to load the plugin of model "
        f"'{_model.name}'.\n"
        "In the meantime, you can try uninstalling the plugin with:\n"
        f"    pip uninstall {get_package_name(_model.value.split('.')[0])}"
    )
    raise ImportError(msg) from e
```

Will fail with the following error:

```
[AutoAPI] Reading files... [ 26%] /home/bart/git/ewatercycle/src/ewatercycle/models.py
Extension error (autoapi.extension):
Handler <function run_autoapi at 0x7f605e290d60> for event 'builder-inited' threw an exception (exception: Inference failed for all members of [<JoinedStr l.46 at 0x7f60437fb260>].)
```

However, modifying the error message to the following seems to be fine:

```py
    pkg_name = get_package_name(_model.value.split(".")[0])
    msg = (
        "An error was raised when trying to load the plugin of model "
        f"'{_model.name}'.\n"
        "In the meantime, you can try uninstalling the plugin with:\n"
        f"    pip uninstall {pkg_name}"
    )
```

This error only showed up today on the readthedocs build, and I was able to reproduce it locally. The change to the code was made 5 months ago and up to this week it did not produce any errors.

<details><summary>pip list</summary>
<p>

Package                       Version
----------------------------- -----------
alabaster                     1.0.0
astroid                       3.3.5
asttokens                     3.0.0
attrs                         24.2.0
babel                         2.16.0
certifi                       2024.8.30
charset-normalizer            3.4.0
click                         8.1.7
comm                          0.2.2
debugpy                       1.8.9
decorator                     5.1.1
docutils                      0.21.2
executing                     2.1.0
fastjsonschema                2.21.1
greenlet                      3.1.1
idna                          3.10
imagesize                     1.4.1
importlib_metadata            8.5.0
ipykernel                     6.29.5
ipython                       8.30.0
jedi                          0.19.2
Jinja2                        3.1.4
jsonschema                    4.23.0
jsonschema-specifications     2024.10.1
jupyter-cache                 1.0.1
jupyter_client                8.6.3
jupyter_core                  5.7.2
markdown-it-py                3.0.0
MarkupSafe                    3.0.2
matplotlib-inline             0.1.7
mdit-py-plugins               0.4.2
mdurl                         0.1.2
myst-nb                       1.1.2
myst-parser                   4.0.0
nbclient                      0.10.1
nbformat                      5.10.4
nest-asyncio                  1.6.0
packaging                     24.2
parso                         0.8.4
pexpect                       4.9.0
pip                           24.2
platformdirs                  4.3.6
prompt_toolkit                3.0.48
psutil                        6.1.0
ptyprocess                    0.7.0
pure_eval                     0.2.3
Pygments                      2.18.0
python-dateutil               2.9.0.post0
PyYAML                        6.0.2
pyzmq                         26.2.0
referencing                   0.35.1
requests                      2.32.3
rpds-py                       0.21.0
ruamel.yaml                   0.18.6
ruamel.yaml.clib              0.2.12
six                           1.16.0
snowballstemmer               2.2.0
Sphinx                        8.1.3
sphinx-autoapi                3.4.0
sphinx-copybutton             0.5.2
sphinx-rtd-theme              3.0.2
sphinxcontrib-applehelp       2.0.0
sphinxcontrib-devhelp         2.0.0
sphinxcontrib-htmlhelp        2.1.0
sphinxcontrib-jquery          4.1
sphinxcontrib-jsmath          1.0.1
sphinxcontrib-qthelp          2.0.0
sphinxcontrib-serializinghtml 2.0.0
SQLAlchemy                    2.0.36
stack-data                    0.6.3
tabulate                      0.9.0
tomli                         2.2.1
tornado                       6.4.2
traitlets                     5.14.3
typing_extensions             4.12.2
urllib3                       2.2.3
wcwidth                       0.2.13
zipp                          3.21.0

</p>
</details>

**Comment by Pierre-Sassoulas** [comment]
Thank you for opening the issue. Might be an exception  that should be caught and handled in sphinx-api. And this particular issue might be fixable in astroid.

**Comment by jacobtylerwalls** [comment]
> Might be an exception that should be caught and handled in sphinx-api.

This sounds right. Can you share the source for the call to astroid?

**Comment by Pierre-Sassoulas** [comment]
I think this is both a downstream bug (as astroid can sometime raise those exceptions during 'normal operation' they need to be handled downstream) but also a bug in astroid, what do you think Jacob ?

**Comment by jacobtylerwalls** [comment]
What version of astroid is installed? This might be fixed by #2622.

**Comment by jacobtylerwalls** [comment]
Thanks. Might you be able to retest against main?

**Comment by BSchilperoort** [comment]
> Thanks. Might you be able to retest against main?

Yeah. It seems fixed for main, still present in 3.3.6.

To reproduce

```sh
git clone --depth 1 https://github.com/eWaterCycle/ewatercycle.git .
git fetch origin --force --prune --prune-tags --depth 50 pull/468/head:external-468
git checkout --force 26b34b1e5d3a778fa26130d6a88834cc915d3ab0
python -m venv .venv
source .venv/bin/activate
python -m pip install --upgrade --no-cache-dir pip setuptools
python -m pip install --upgrade --no-cache-dir sphinx
python -m pip install --exists-action=w --no-cache-dir -r docs/requirements.txt
cd docs
make html
```
gives error

```sh
pip list | grep astroid
>>> astroid                       3.3.6
```

```sh
pip install --no-deps https://github.com/pylint-dev/astroid/archive/refs/heads/main.zip
make html
```
gives no error anymore.

**Comment by jacobtylerwalls** [comment]
Great, thanks for confirming. I'll close as fixed and add to the 4.0 milestone.
